components/subcomponents/movieNight/upcomingMovie.py

```diff
@@ -0,0 +1,211 @@
+
+# Sends reminder to upcoming_host every noon until a movie from suggestion-list is picked
+def start_pick_reminder():
+    if __should_send_reminder():
+        pickReminderTask.start()
+
+
+# Updates the description of ScheduledEvent
+def update_event_description(is_command):
+    updateEventDescriptionTask.start(is_command)
+
+
+# Check to see if host should be reminded to pick a movie for movie night
+# Only set up reminders if:
+#   1) host has been picked, but not the movie
+#   2) ScheduledEvent MOVIE_EVENT_NAME exists
+#   3) ScheduledEvent takes place in the future
+def __should_send_reminder():
+    event: discord.ScheduledEvent | None = ut.get_movie_event()
+    db = shelve.open(UPCOMING_MOVIE_NIGHT_DB_PATH)
+    upcoming_host_name: str | None = db.get("upcoming_host_name")
+    upcoming_movie: Movie | None = db.get("upcoming_movie")
+    db.close()
+
+    return upcoming_host_name and not upcoming_movie and event and datetime.now(timezone.utc) <= event.start_time
+
+
+# Coroutine to remind the host to pick the movie
+# Runs in the background as a task in the following situations:
+#   1) every time an upcoming_host has been set
+#   2) every time the application restarts
+async def __remind_host_coroutine():
+    try:
+        while True:
+            # Set reminder for tomorrow at noon
+            next_noon = (datetime.now() + timedelta(days=1)).replace(hour=12, minute=0, second=0, microsecond=0)
+            delta = next_noon - datetime.now()
+            await asyncio.sleep(delta.total_seconds())
+
+            # Don't send reminders if it is unnecessary
+            if not __should_send_reminder():
+                return
+
+            # Send reminder since movie has not been picked
+            event: discord.ScheduledEvent | None = ut.get_movie_event()
+            db = shelve.open(UPCOMING_MOVIE_NIGHT_DB_PATH)
+            upcoming_host_name: str | None = db.get("upcoming_host_name")
+            db.close()
+
+            embed = discord.Embed(colour=ut.embed_colour["MOVIE_NIGHT"])
+            host_str = ut.get_member_str(upcoming_host_name)
+            embed.title = f"{host_str}, please select the upcoming movie!"
+            embed.description = f"Please select a movie before {ut.convert_to_ottawa_time(event.start_time)}."
+            await ut.send_message(ut.get_channel(ut.env["MOVIE_CHANNEL"]), host_str, embed)
+    except Exception as e:
+        await ut.send_message(ut.get_channel(ut.env["MOVIE_CHANNEL"]), "Error with Movie Reminder: " + str(e))
+
+
+# Coroutine to update the description of ScheduledEvent
+# Runs in the background as a task in the following situations:
+#   1) when a host is picked or reset for the upcoming movie night
+#   2) when a host picks a movie for the upcoming movie night
+#   3) when the bot restarts
+#   4) when ScheduledEvent is updated
+# NOTE: Will only send an edit request if title or description has changed to save on bandwidth
+async def __update_description_coroutine(is_command):
```

Comment:
question: What is the benefit of this being a coroutine?

Reply:
Having coroutines prevents bugs related to hogging performance. Example: "2024-10-20 07:39:33 WARNING discord.gateway Shard ID None voice heartbeat blocked for more than 10 seconds"